utils/python/CIME/XML/machines.py

```diff
@@ -27,9 +27,9 @@ def __init__(self, infile=None, files=None, machine=None):
         if infile is None:
             if files is None:
                 files = Files()
-            infile = files.get_value("MACHINES_SPEC_FILE")
+            infile = files.get_value("MACHINES_SPEC_FILE", resolved=False)
             self.machines_dir = os.path.dirname(infile)
-
+            infile = files.get_resolved_value(infile)
```

Comment:
I don't understand this change. Why is calling get_resolved_value by hand better than resolved=True?